src/Simplex/Chat.hs

```diff
@@ -3542,13 +3542,23 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
           let ExtMsgContent _ _ itemTTL live_ = mcExtMsgContent mc
               timed_ = rcvGroupCITimed gInfo itemTTL
               live = fromMaybe False live_
-          file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
-          ChatItem {formattedText} <- newChatItem (CIRcvMsgContent content) (snd <$> file_) timed_ live
-          autoAcceptFile file_
-          let g = groupName' gInfo
-          whenGroupNtfs user gInfo $ do
-            showMsgToast ("#" <> g <> " " <> c <> "> ") content formattedText
-            setActive $ ActiveG g
+          -- check if message moderation event was received ahead of message
+          withStore' (\db -> findModeration db user gInfo memberId sharedMsgId_) >>= \case
+            Nothing -> do
+              file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
+              ChatItem {formattedText} <- newChatItem (CIRcvMsgContent content) (snd <$> file_) timed_ live
+              autoAcceptFile file_
+              let g = groupName' gInfo
+              whenGroupNtfs user gInfo $ do
+                showMsgToast ("#" <> g <> " " <> c <> "> ") content formattedText
+                setActive $ ActiveG g
+            Just (moderationId, moderatorMember, createdByMsgId, moderatedAtTs) -> do
+              unless (groupFeatureAllowed SGFFullDelete gInfo) $ do
+                file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
```

Comment:
if this item already moderated, we probably don't need to process file invitation at all? There is (should be) no way to receive moderated files, is there?

Reply:
it doesn't receive it, but it creates records for proper chat item retrieval later on - it can be revealed